File: MusicFestival/Repository/OfficeEmployeeDBRepository.cs
using MPP_MusicFestival.Domain;
using MPP_MusicFestival.Repository;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using log4net;

namespace MusicFestival.Repository
{
    class OfficeEmployeeDBRepository : OfficeEmployeeRepository
    {

        private static readonly ILog log = LogManager.GetLogger("OfficeEmployeeDBRepository");

        IDictionary<String, string> props;

        public OfficeEmployeeDBRepository(IDictionary<String, string> props)
        {
            log.Info("Creating OfficeEmployeeDBRepository ");
            this.props = props;
        }

        public void delete(long id)
        {
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from OfficeEmployee where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();
                //if (dataR == 0)
                //  throw new RepositoryException("No task deleted!");
            }
        }

        public IEnumerable<OfficeEmployee> findAll()
        {
            IDbConnection con = DBUtils.getConnection(props);
            IList<OfficeEmployee> officeEmployees = new List<OfficeEmployee>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from OfficeEmployee";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        long id = dataR.GetInt32(0);
                        String first_name = dataR.GetString(1);
                        String last_name = dataR.GetString(2);
                        long CNP = dataR.GetInt32(3);
                        String username = dataR.GetString(4);
                        String password = dataR.GetString(5);
                        OfficeEmployee officeEmployee = new OfficeEmployee(id, first_name, last_name, CNP, username, password);

                        officeEmployees.Add(officeEmployee);
                    }
                }
            }

            return officeEmployees;
        }

        public OfficeEmployee findOne(long id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from OfficeEmployee where id=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idO = dataR.GetInt32(0);
                        String first_name = dataR.GetString(1);
                        String last_name = dataR.GetString(2);
                        long CNP = dataR.GetInt32(3);
                        String username = dataR.GetString(4);
                        String password = dataR.GetString(5);
                        OfficeEmployee officeEmployee = new OfficeEmployee(idO, first_name, last_name, CNP, username, password);

                        log.InfoFormat("Exiting findOne with value {0}", officeEmployee);
                        return officeEmployee;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public long getId(string username)
        {
            IDbConnection con = DBUtils.getConnection(props);

            long N = 0;
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from OfficeEmployee where username=@username";
                var paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = username;
                comm.Parameters.Add(paramUsername);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idO = dataR.GetInt32(0);
                        
                        log.InfoFormat("Exiting getId with value {0}", idO);
                        return idO;
                    }
                }
            }
            return N;
        }

        public string getPassword(long id)
        {
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from OfficeEmployee where id=@id";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        string password = dataR.GetString(5);

                        log.InfoFormat("Exiting getPassword with value {0}", password);
                        return password;
                    }
                }
            }
            return null;
        }
        public long getId(long CNP)
        {
            IDbConnection con = DBUtils.getConnection(props);

            long N = 0;
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from OfficeEmployee where CNP=@CNP";
                var paramCNP = comm.CreateParameter();
                paramCNP.ParameterName = "@CNP";
                paramCNP.Value = CNP;
                comm.Parameters.Add(paramCNP);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idO = dataR.GetInt32(0);

                        log.InfoFormat("Exiting getId with value {0}", idO);
                        return idO;
                    }
                }
            }
            return N;
        }

        public void save(OfficeEmployee entity)
        {
            var con = DBUtils.getConnection(props);

            if (getId(entity.CNP) != 0)
                log.InfoFormat("This entity already exists");
            else
            {

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "insert into OfficeEmployee(first_name, last_name, CNP, username, password) values (@first_name, @last_name, @CNP, @username, @password)";
                    var paramfName = comm.CreateParameter();
                    paramfName.ParameterName = "@first_name";
                    paramfName.Value = entity.firstName;
                    comm.Parameters.Add(paramfName);

                    var paramlName = comm.CreateParameter();
                    paramlName.ParameterName = "@last_name";
                    paramlName.Value = entity.lastName;
                    comm.Parameters.Add(paramlName);

                    var paramCNP = comm.CreateParameter();
                    paramCNP.ParameterName = "@CNP";
                    paramCNP.Value = entity.CNP;
                    comm.Parameters.Add(paramCNP);

                    var paramUsername = comm.CreateParameter();
                    paramUsername.ParameterName = "@username";
                    paramUsername.Value = entity.username;
                    comm.Parameters.Add(paramUsername);

                    var paramPassw = comm.CreateParameter();
                    paramPassw.ParameterName = "@password";
                    paramPassw.Value = entity.password;
                    comm.Parameters.Add(paramPassw);

                    var result = comm.ExecuteNonQuery();
                    //if (result == 0)
                    //  throw new RepositoryException("No task added !");
                }
            }
        }

        public void update(long id, OfficeEmployee entity)
        {
            throw new NotImplementedException();
        }
    }
}

File: MPP_MusicFestival_Java/FestivalPersistence/src/main/java/repository/database/OfficeEmployeeDBRepository.java
package repository.database;

import domain.OfficeEmployee;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import repository.OfficeEmployeeRepository;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class OfficeEmployeeDBRepository implements OfficeEmployeeRepository {

    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public OfficeEmployeeDBRepository(Properties props) {
        logger.info("Initializing OfficeEmployeeDBRepository with properties: {} ", props);
        dbUtils = new JdbcUtils(props);
    }


    @Override
    public Long getId(String username) {
        logger.traceEntry("finding office employee id with username {} ", username);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("select * from OfficeEmployee where username=?")) {
            preStmt.setString(1, username);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    long id = result.getLong("id");

                    logger.traceExit(id);
                    return id;
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit("No office employee found with username {}", username);
        return null;
    }

    @Override
    public String getPassword(Long id) {
        logger.traceEntry("finding office employee password with id {} ", id);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("select * from OfficeEmployee where id=?")) {
            preStmt.setLong(1, id);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    String username = result.getString("password");

                    logger.traceExit(username);
                    return username;
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit("No office employee password found with id {}", id);
        return null;
    }

    @Override
    public Long getId(Long CNP){
        logger.traceEntry("finding office employee id with CNP {} ", CNP);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("select * from OfficeEmployee where CNP=?")) {
            preStmt.setLong(1, CNP);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    long id = result.getLong("id");

                    logger.traceExit(id);
                    return id;
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit("No office employee found with CNP {}", CNP);
        return null;
    }


    @Override
    public void save(OfficeEmployee entity) {
        logger.traceEntry("saving office employee {} ", entity);
        Connection con = dbUtils.getConnection();

        if (getId(entity.getCNP()) != null)
            logger.error("This entity already exists");
        else {
            try (PreparedStatement preStmt = con.prepareStatement("insert into OfficeEmployee(first_name, last_name, CNP, username, password) values (?,?,?,?,?)")) {
                preStmt.setString(1, entity.getFirstName());
                preStmt.setString(2, entity.getLastName());
                preStmt.setLong(3, entity.getCNP());
                preStmt.setString(4, entity.getUsername());
                preStmt.setString(5, entity.getPassword());
                preStmt.executeUpdate();
            } catch (SQLException ex) {
                logger.error(ex);
                System.out.println("Error DB " + ex);
            }
            logger.traceExit();
        }
    }

    @Override
    public void delete(Long aLong) {
        logger.traceEntry("deleting office employee with {}", aLong);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("delete from OfficeEmployee where id=?")) {
            preStmt.setLong(1, aLong);
            preStmt.executeUpdate();
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Long aLong, OfficeEmployee entity) {
    }

    @Override
    public OfficeEmployee findOne(Long aLong) {
        logger.traceEntry("finding office employee with id {} ", aLong);
        Connection con = dbUtils.getConnection();

        try (PreparedStatement preStmt = con.prepareStatement("select * from OfficeEmployee where id=?")) {
            preStmt.setLong(1, aLong);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    long id = result.getLong("id");
                    String first_name = result.getString("first_name");
                    String last_name = result.getString("last_name");
                    long CNP = result.getLong("CNP");
                    String username = result.getString("username");
                    String password = result.getString("password");

                    OfficeEmployee officeEmployee = new OfficeEmployee(id, first_name, last_name, CNP, username, password);
                    logger.traceExit(officeEmployee);
                    return officeEmployee;
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        }
        logger.traceExit("No office employee found with id {}", aLong);
        return null;
    }

    @Override
    public List<OfficeEmployee> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<OfficeEmployee> officeEmployees = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from OfficeEmployee")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    long id = result.getLong("id");
                    String first_name = result.getString("first_name");
                    String last_name = result.getString("last_name");
                    long CNP = result.getLong("CNP");
                    String username = result.getString("username");
                    String password = result.getString("password");

                    OfficeEmployee officeEmployee = new OfficeEmployee(id, first_name, last_name, CNP, username, password);
                    officeEmployees.add(officeEmployee);
                }
            }

        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return officeEmployees;
    }
}
